src/transport/grpc/mod.rs

```diff
@@ -0,0 +1,837 @@
+        _mapping: MappingRef<'_, &Interface>,
+        payload: Self::Payload,
+    ) -> Result<(AstarteType, Option<Timestamp>), crate::Error> {
+        let astarte_data_type::Data::AstarteIndividual(individual_data) = payload.data else {
+            return Err(crate::Error::from(
+                GrpcTransportError::DeserializationExpectedIndividual,
+            ));
+        };
+
+        Ok((individual_data.try_into()?, payload.timestamp))
+    }
+
+    fn deserialize_object(
+        &self,
+        _object: ObjectRef,
+        _path: &MappingPath<'_>,
+        payload: Self::Payload,
+    ) -> Result<(HashMap<String, AstarteType>, Option<Timestamp>), crate::Error> {
+        let astarte_data_type::Data::AstarteObject(object) = payload.data else {
+            return Err(crate::Error::from(
+                GrpcTransportError::DeserializationExpectedObject,
+            ));
+        };
+
+        Ok((
+            map_values_to_astarte_type(object.object_data)?,
+            payload.timestamp,
+        ))
+    }
+}
+
+#[async_trait]
+impl Register for Grpc {
+    async fn add_interface<S>(
+        &self,
+        device: &SharedDevice<S>,
+        _added_interface: &str,
+    ) -> Result<(), crate::Error>
+    where
+        S: PropertyStore,
+    {
+        let data = NodeData::try_from_unlocked(self.uuid, &*device.interfaces.read().await)?;
+
+        self.try_detach_attach(data)
+            .await
+            .map_err(crate::Error::from)
+    }
+
+    async fn remove_interface(
+        &self,
+        interfaces: &Interfaces,
+        _removed_interface: Interface,
+    ) -> Result<(), crate::Error> {
+        let data = NodeData::try_from_unlocked(self.uuid, interfaces)?;
+
+        self.try_detach_attach(data)
+            .await
+            .map_err(crate::Error::from)
+    }
+}
+
+#[async_trait]
+impl Disconnect for Grpc {
+    async fn disconnect(mut self) -> Result<(), crate::Error> {
+        if let Some(SharedGrpc {
+            ref mut client,
+            stream: _stream,
+            uuid,
+        }) = Arc::get_mut(&mut self.shared)
+        {
+            Self::try_detach(client.get_mut(), uuid.to_string())
+                .await
+                .map_err(|e| e.into())
+        } else {
+            Err(GrpcTransportError::GracefulClose.into())
+        }
+    }
+}
+
+/*
+/// Implemented to correctly detach the device from the message hub
+///
+/// This implementation currently blocks the current thread to wait
+/// for the complete disconnection from the message hub.
+impl Drop for Grpc {
+    fn drop(&mut self) {
+    }
+}
+*/
+
+/// Internal struct holding the received grpc message
+#[derive(Debug, PartialEq)]
+pub(crate) struct GrpcReceivePayload {
+    data: astarte_data_type::Data,
+    timestamp: Option<Timestamp>,
+}
+
+impl GrpcReceivePayload {
+    pub(crate) fn new(data: astarte_data_type::Data, timestamp: Option<Timestamp>) -> Self {
+        Self { data, timestamp }
+    }
+}
+
+pub struct GrpcConfig {
```

Comment:
```suggestion
#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
pub struct GrpcConfig {
```

Reply:
And probably display since it's bug, but not really necessary

Reply:
Avoided making it displayable but I derived the other traits like you suggested.